econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -214,11 +215,26 @@ def _sanitize(obj):
             raise ValueError(f"Could not sanitize input {obj}")
 
 
+# Convert SingleTreeInterpreter to a python dictionary
+def _tree_interpreter_to_dict(interp, features, leaf_data=lambda t, n: {}):
+    tree = interp.tree_model_.tree_
+
+    def recurse(node_id):
+        if tree.children_left[node_id] == _tree.TREE_LEAF:
+            return {'leaf': True, 'n_samples': tree.n_node_samples[node_id], **leaf_data(tree, node_id)}
+        else:
+            return {'leaf': False, 'feature': features[tree.feature[node_id]], 'threshold': tree.threshold[node_id],
+                    'left': recurse(tree.children_left[node_id]),
+                    'right': recurse(tree.children_right[node_id])}
+
+    return recurse(0)
+
+
 # named tuple type for storing results inside CausalAnalysis class;
 # must be lifted to module level to enable pickling
 _result = namedtuple("_result", field_names=[
     "feature_index", "feature_name", "feature_baseline", "feature_levels", "hinds",
-    "X_transformer", "W_transformer", "estimator", "global_inference"])
+    "X_transformer", "W_transformer", "estimator", "global_inference", "treatment_value"])
 
 
 class CausalAnalysis:
```

Comment:
Shouldn't we also have a "random_state" input to the object, so that people will be able to reproduce results by rerunning and so that this random_state if not None, is propagated to all parts of the process (e.g. model selection and CATE estimation)

Reply:
Yes, done.